test/endpoints/bar/bar_metrics_test.go

```diff
@@ -116,26 +116,27 @@ func TestCallMetrics(t *testing.T) {
 	for k, v := range endpointTags {
 		statusTags[k] = v
 	}
-	histogramTags := map[string]string{
-		m3.DefaultHistogramBucketName:   "0-10ms", // TODO(argouber): Remove the ugly hardcoding
-		m3.DefaultHistogramBucketIDName: "0001",
-	}
-	for k, v := range statusTags {
-		histogramTags[k] = v
-	}
+
 	key := tally.KeyForPrefixedStringMap("endpoint.request", endpointTags)
 	assert.Contains(t, metrics, key, "expected metric: %s", key)
 	assert.Equal(t, int64(1), metrics[key].Value.Count)
 
 	key = tally.KeyForPrefixedStringMap("endpoint.latency", statusTags)
 	assert.Contains(t, metrics, key, "expected metric: %s", key)
 	value := metrics[key].Value.Timer
-	assert.True(t, value > 1000, "expected latency > 1000 nano seconds")
-	assert.True(t, value < 10*1000*1000, "expected latency < 10 milli seconds")
 
-	key = tally.KeyForPrefixedStringMap("endpoint.latency-hist", histogramTags)
-	assert.Contains(t, metrics, key, "expected metric: %s", key)
-	assert.Equal(t, int64(1), metrics[key].Value.Count)
+	assert.True(t, value > 1000, fmt.Sprintf("expected latency > 1000 nano seconds but found to be: %d", value))
+	assert.True(t, value < 50*1000*1000, fmt.Sprintf("expected latency < 50 milli seconds but found to be: %d", value))
+
+	key = "endpoint.latency-hist"
+	keyFound := false
+	for metricKeyName := range metrics {
+		if strings.Contains(metricKeyName, key) {
+			assert.Equal(t, int64(1), metrics[metricKeyName].Value.Count, fmt.Sprintf("key: %s, metric: %v\n", key, metrics[key]))
```

Comment:
Should we add a nil check for metrics[metricKeyName] ?

Reply:
Addressed.